tests/nataliia_romanchuk/homework_5/test_dicts.py

```diff
@@ -0,0 +1,121 @@
+# DICTS
+import pytest
+
+some_dict = {"a": 123, "b": 456, "c": 789}
+some_dict_2 = {"c": 789, "a": 123, "b": 456}
+some_dict_3 = dict.fromkeys("abc")
+wierd_dict = {"a": 1, 2: "b",
+              None: "abc",
+              (1, "a"): "some",
+              True: "kek",
+              frozenset((1, 2)): "lol"}
+
+
+# is container
+def test_dict_comparison():
+    assert some_dict == some_dict_2
+    assert "a" in some_dict
+    assert len(some_dict) == 3
+    assert list(some_dict_2) == ['c', 'a', 'b']
+
+
+def test_dict_methods():
+    assert some_dict["a"] == 123
+    assert some_dict.get("a") == 123
+    assert some_dict["a"] == some_dict.get("a")
+    assert some_dict.get("invalid") == None
+    assert some_dict.get("invalid", "default_val") == 'default_val'
+
+
+@pytest.mark.parametrize("some", [{"a": 1, "b": 2}])
+def test_dict_update(some):
+    some.update({"b": "b", "c": "c"})
+    assert some['a'] == 1
+    assert some['b'] != 2
+    assert some['b'] == 'b'
+    assert some['c'] == 'c'
+
+
+@pytest.mark.parametrize("some", [{"a": 1, "b": 2}])
+def test_dict_copy(some):
+    sopy_of_some = some.copy()
+    assert sopy_of_some == some
+    assert some['a'] == 1
+    assert some['b'] == 2
+
+    some.clear()
+    assert some == {}
+
+
+@pytest.mark.parametrize("some", [{"a": 1, "b": 2}])
+def test_dict_methods(some):
+    item = some.pop("a")
+    assert item == 1
+    assert 'a' not in some
+
+    item = some.pop("invalid", "default_val")
+    assert item == "default_val"
+
+    item = some.setdefault("a", "default_for_a")
+    assert "a" in some
+    assert item == some['a']
+
+    item = some.setdefault("a", "default_for_a")
+    assert "a" in some
+    assert some["a"] == "default_for_a"
+    assert item == "default_for_a"
+
+    item = some.setdefault("a", "other_default_for_a")
+    assert "a" in some
+    assert "a" in item
+    assert some['a'] == "default_for_a"
+    assert item == "default_for_a"
```

Comment:
better to divide test into sepparate tests

Reply:
added 1 more test for default